### ISSUE TITLE
[BUG] Type map for single element

### ISSUE THREAD

**Issue by romankempt** [deepmodeling/deepmd-kit#1931]
### Bug summary

[Here](https://github.com/deepmodeling/deepmd-kit/blob/c39339247a8a5a254fa7509be898771ab00498e9/deepmd/utils/data.py#L546-L547) in the deepmd-kit/data.py module, these two lines

```
        if self.type_map is not None:
            assert(len(self.type_map) >= max(self.atom_type)+1)
```

raise an error if there are less entries in the type_map than the largest atom type. E.g., we have a bulk Pt structure with 4 atoms:
```
Data Summary
Labeled System
-------------------
Frame Numbers      : 5
Atom Numbers       : 4
Including Virials  : Yes
Element List       :
-------------------
Pt
4
[0 0 0 0]
```

Where the type_map only has the entry "Pt" and the atom_type are [0,0,0,0]. Is this intended?

### DeePMD-kit Version

v2.1.4

### TensorFlow Version

2.9.0

### How did you download the software?

conda

### Input Files, Running Commands, Error Log, etc.

[example.zip](https://github.com/deepmodeling/deepmd-kit/files/9599359/example.zip)

### Steps to Reproduce

The example dataset can be loaded via:

```python
from deepmd.utils.data import DeepmdData
d = DeepmdData("example")
```

which triggers the error.

**Comment by njzjz** [comment]
> which triggers the error.

Failed to reproduce:
https://gist.github.com/njzjz/7111be76ff93d73da8f1ee2b6ae6b7ba

**Comment by romankempt** [comment]
Sorry!

The error is triggered here: 
[example.zip](https://github.com/deepmodeling/deepmd-kit/files/9607651/example.zip)

I've tried to define Pt globally as type 0 and Se globally as type 1.

The error is avoided when changing the `type_map.raw` from:
`Se` to `Pt Se`.

But this type map is not automatically generated when dumping the LabeledSystem to the `deepmd/npy` format, even though the LabeledSystem has the type map. Is there a way to force writing out a specific type map?

```python
    type_map = ["Pt", "Se"]

    data = {"atom_numbs": nums,
            "atom_names": names,
            "atom_types": types,
            "orig": np.array([0,0,0]),
            "cells": np.asarray(cell),
            "coords": np.asarray(positions),
            "nopbc": False,
            "forces": np.asarray(forces),
            "energies": np.asarray(energy),
            "virials": np.asarray(stress)
            }
    ls = LabeledSystem(data=data, type_map=type_map)
    ls.to("deepmd/npy", "path", set_size = l.get_nframes())
```

**Comment by njzjz** [comment]
Indeed, the new file is incorrect. The index of `type.raw` should map to the names in `type_map.raw`. See [documentation](https://docs.deepmodeling.com/projects/deepmd/en/v2.1.4/data/system.html).

Your `type.raw` has index 1 (meaning there should be at least two elements) but `type_map.raw` has only one element.
